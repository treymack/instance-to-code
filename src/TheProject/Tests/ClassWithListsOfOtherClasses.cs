using System;
using System.Collections.Generic;
using System.Linq;
using ApprovalTests;
using NUnit.Framework;

namespace TheProject.Tests
{
    [TestFixture]
    public class ClassWithListsOfOtherClasses
    {
        [Test]
        public void CanScriptToCode()
        {
            var instance = new ClassListClass
            {
                Users = new List<User2>
                {
                    new User2
                    {
                        Name = "tmack",
                        Address = new Address2{
                            AddressLine1 = "line1",
                        },
                        Personality = new Personality2
                        {
                            PersonalityDescriptor = "awesome",
                        },
                    },
                    new User2
                    {
                        Personality = new Personality2
                        {
                            PersonalityDescriptor = "blank",
                        },
                    }
                },
                Addresses = new List<Address2> { },
            };

            var code = ITC.Instance.ToCode(instance);

            Approvals.Verify(code);
        }
    }

    public class ClassListClass
    {
        public List<User2> Users { get; set; }

        public List<Address2> Addresses { get; set; }
    }
}
