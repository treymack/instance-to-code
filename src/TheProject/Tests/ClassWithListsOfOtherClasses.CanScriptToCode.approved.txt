new TheProject.Tests.ClassListClass()
{
Addresses = new System.Collections.Generic.List<TheProject.Tests.Address2>
{},
Users = new System.Collections.Generic.List<TheProject.Tests.User2>
{new TheProject.Tests.User2()
{
Address = new TheProject.Tests.Address2()
{
AddressLine1 = "line1",
},
ByteArray = null, 
Name = "tmack",
Personality = new TheProject.Tests.Personality2()
{
PersonalityDescriptor = "awesome",
},
}, 
new TheProject.Tests.User2()
{
Address = null,
ByteArray = null, 
Name = null,
Personality = new TheProject.Tests.Personality2()
{
PersonalityDescriptor = "blank",
},
}, 
},
}